opm/simulators/linalg/setupPropertyTree_impl.hpp

```diff
@@ -0,0 +1,112 @@
+/*
+  Copyright 2019 SINTEF Digital, Mathematics and Cybernetics.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <config.h>
+
+#include <opm/simulators/linalg/setupPropertyTree.hpp>
+
+#include <boost/version.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+namespace Opm
+{
+
+/// Set up a property tree intended for FlexibleSolver by either reading
+/// the tree from a JSON file or creating a tree giving the default solver
+/// and preconditioner. If the latter, the parameters --linear-solver-reduction,
+/// --linear-solver-maxiter and --linear-solver-verbosity are used, but if reading
+/// from file the data in the JSON file will override any other options.
+template<class TypeTag>
+boost::property_tree::ptree
+setupPropertyTree(const FlowLinearSolverParameters& p)
+{
+    boost::property_tree::ptree prm;
+    if (p.linear_solver_configuration_ == "file") {
+#if BOOST_VERSION / 100 % 1000 > 48
```

Comment:
Logic still looks wrong here, and error messages should be adjusted. Is there something I don't see?

Reply:
Now i get it.

Reply:
The logic is correct, but the message is wrong.

Reply:
By "the logic" I mean what happens when the BOOST_VERSION is too low (i.e. the above is false). Then we have
```
    if (p.linear_solver_configuration_ == "file") {
    }
    else
    {
        std::string conf =  p.linear_solver_configuration_;
        ...
```

Reply:
We aree now throwing in this case.